postgres-external-storage/src/main/java/com/netflix/conductor/postgres/storage/PostgresPayloadStorage.java

```diff
@@ -104,17 +107,19 @@ public void upload(String key, InputStream payload, long payloadSize) {
      *
      * @param key the PostgreSQL key of the object
      * @return an input stream containing the contents of the object. Caller is expected to close
-     *     the input stream.
+     * the input stream.
      */
     @Override
     public InputStream download(String key) {
         InputStream inputStream;
         try (Connection conn = postgresDataSource.getConnection();
-                PreparedStatement stmt =
-                        conn.prepareStatement("SELECT data FROM " + tableName + " WHERE id = ?")) {
+             PreparedStatement stmt = conn.prepareStatement("SELECT data FROM " + tableName + " WHERE id = ?")) {
             stmt.setString(1, key);
             ResultSet rs = stmt.executeQuery();
-            rs.next();
+            if (!rs.next()) {
+                LOGGER.debug("External PostgreSQL data with this ID: {} does not exist", key);
+                return new ByteArrayInputStream(DEFAULT_MESSAGE_TO_USER.getBytes());
```

Comment:
Small problem here: charset not specified. We should use utf8 everywhere ... but the problem is that this service just works with bytes in upload/download ... so we don't know what charset is used so using utf8 as default here (Explicitly specified) would be good but not ideal

Reply:
... maybe we should make the default value configurable ? higher layer would pass the default message as bytes (already serialized with desired encoding)

Reply:
We don't need to add any encoding here because at the level above it is already in place:        
```
 try (InputStream inputStream = externalPayloadStorage.download(path)) {
            return objectMapper.readValue(
                    IOUtils.toString(inputStream, StandardCharsets.UTF_8), Map.class);
```